frontend/packages/kubevirt-plugin/src/components/create-vm/dev-console/dev-console-create-vm-drawer.tsx

```diff
@@ -0,0 +1,130 @@
+import { isTemplateSourceError, TemplateSourceStatus } from '../../../statuses/template/types';
+import { SourceDescription } from '../../vm-templates/vm-template-source';
+
+const getTemplateCatalogItem = (
+  template: TemplateKind,
+  sourceStatus: TemplateSourceStatus,
+  isUserProvided: boolean,
+) => ({
+  obj: template,
+  tileProvider: getTemplateProvider(template),
+  longDescription: isCommonTemplate(template)
+    ? 'This template is provided by Red Hat. The Boot source is also maintained by Red Hat.'
+    : 'This template is provider by a User',
+  createLabel: 'Select Template',
+  href: '/',
+  customDescriptions: (
+    <>
+      <SectionHeading text="Details" />
+      <Form>
+        <FormRow title="Namespace" fieldId="dev-console-vm-namespace">
+          {getNamespace(template)}
+        </FormRow>
+        {isUserProvided && (
+          <FormRow title="Operating System" fieldId="dev-console-vm-os">
+            {getOperatingSystemName(template) || 'Not available'}
+          </FormRow>
+        )}
+        <FormRow title="Type" fieldId="dev-console-vm-workload">
+          {getWorkloadProfile(template) || 'Not available'}
+        </FormRow>
+        <FormRow
+          fieldId="dev-console-vm-flavor"
+          title="Flavor"
+          help="You will be able to change the flavor after selecting this template"
+        >
+          {getTemplateFlavorDesc(template)}
+        </FormRow>
+        <FormRow title="Storage" fieldId="dev-console-vm-storage">
+          {getTemplateSizeRequirement(template, sourceStatus)}
+        </FormRow>
+        <FormRow title="Boot source" fieldId="dev-console-vm-boot-source">
+          <SourceDescription
+            sourceStatus={!isTemplateSourceError(sourceStatus) && sourceStatus}
+            template={template}
+          />
+        </FormRow>
+      </Form>
+    </>
+  ),
+});
+
+type DevConsoleCreateVmDrawerProps = {
+  selectedTemplate: TemplateItem;
+  sourceStatus: TemplateSourceStatus;
+  namespace: string;
+  onClose: () => void;
+};
+
+export const DevConsoleCreateVmDrawer: React.FC<DevConsoleCreateVmDrawerProps> = ({
+  selectedTemplate,
+  sourceStatus,
+  namespace,
+  onClose,
+}) => {
+  const template = selectedTemplate?.variants?.[0];
+
+  return (
+    <Modal
+      className="co-catalog-page__overlay co-catalog-page__overlay--right"
+      aria-label="kv-dev-console-create-vm"
+      header={
+        template && (
+          <>
+            <CatalogItemHeader
+              title={getTemplateName(template)}
+              vendor={`Provided by ${getTemplateProvider(template)}`}
+              iconImg={getTemplateOSIcon(template)}
+            />
+            <div className="co-catalog-page__overlay-actions">
+              <Link
+                className="pf-c-button pf-m-primary co-catalog-page__overlay-action"
+                to={`/catalog/create-vm?namespace=${namespace}&common-template=${template?.metadata?.name}`}
```

Comment:
you will also need namespace for user templates. I'd suggest something like

```suggestion
                to={`/catalog/create-vm?namespace=${namespace}&template=${template?.metadata?.name}&templateNs=${template?.metadata?.namespace}`}
```

maybe use `URLSearchParams` ?